### PR TITLE
feat(checkpoint): introduce new resource checkpoint

### PR DESCRIPTION
**Pull request by chicharr**
## Related Issues
https://github.com/adobe/helix-rum-js/issues/103

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -134,3 +134,12 @@ document.querySelectorAll('form').forEach((form) => {
 
 window.addEventListener('visibilitychange', ((event) => sampleRUM.leave(event)));
 window.addEventListener('pagehide', ((event) => sampleRUM.leave(event)));
+
+const observer = new PerformanceObserver((list) => {
+  list.getEntries()
+    .filter((entry) => (entry.responseStatus < 400) && new URL(entry.name).pathname.match('.*(\\.plain\\.html|\\.json)$'))
+    .forEach((entry) => {
+      sampleRUM('resource', { source: entry.name, target: Math.round(entry.duration) });
```
Comment (trieloff) [inline src/index.js:142]:
```suggestion
      sampleRUM('loadresource', { target: entry.name });
```

Comment (trieloff) [reply src/index.js:142]: - `loadresource` seems to be more in line with `viewblock` and `viewmedia`
- `target` is typically URL-shaped
- I think the performance numbers are less important

**Review by trieloff — APPROVED** [review]
Very cool idea to use `PerformanceObserver` for this. I'd jiggle the checkpoint name and source a bit.

File: src/index.js [hunk 2]
```diff
@@ -134,3 +134,12 @@ document.querySelectorAll('form').forEach((form) => {
 
 window.addEventListener('visibilitychange', ((event) => sampleRUM.leave(event)));
 window.addEventListener('pagehide', ((event) => sampleRUM.leave(event)));
+
+const observer = new PerformanceObserver((list) => {
+  list.getEntries()
+    .filter((entry) => (entry.responseStatus < 400) && new URL(entry.name).pathname.match('.*(\\.plain\\.html|\\.json)$'))
```
[the remark this answers is not in the document]
Comment (trieloff) [reply src/index.js:140]: In addition `entry.responseStatus` is not universally supported, so we should only compare it if it is not `undefined`.

From a code perspective, I find chained `.filter().filter()` more readable than a `&&` in the boolean expression, especially when it compares unrelated aspects.

File: src/index.js (hunk 1 above)
Comment (chicharr) [reply src/index.js]: Changed name to `loadresource`.

> I think the performance numbers are less important
The issue description mentions "timing". @davidnuescheler are you interested in the duration of the resource load? or can we get rid of it?

File: src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chicharr) [reply src/index.js]: > https://github.com/adobe/helix-rum-js/issues/103#issuecomment-1774832246
> 
> following this we might want to include resources without a path name at all.

@davidnuescheler ^^ what are your thoughts on this.

**Review by davidnuescheler — APPROVED** [review]

**Pull request merged**